actionpack/lib/action_controller/metal/strong_parameters.rb

```diff
@@ -231,6 +231,9 @@ class Parameters
     #   params.permitted?  # => true
     #   Person.new(params) # => #<Person id: nil, name: "Francesco">
     def initialize(parameters = {})
+      unless parameters.respond_to?(:with_indifferent_access)
```

Comment:
Should we just `rescue NoMethodError` instead? Code is a bit less clear, but `respond_to?` is not free and this is a luke-warm code path.